patches/0116-lavfi-add-LibTorch-backend-multi-frame-inference-sup.patch

```diff
@@ -0,0 +1,402 @@
+From ae22a9bf5be700c9d6f1b8462ba85c45874a2654 Mon Sep 17 00:00:00 2001
+From: Ting Fu <ting.fu@intel.com>
+Date: Mon, 18 Jul 2022 10:13:04 +0800
+Subject: [PATCH] lavfi: add LibTorch backend multi-frame inference support
+
+Dnn models may take multiple frames as one input, so add "nb_input"
+option to control input frame number. Use queue to buffer the input
+frames and send to model at once.
+
+Signed-off-by: Ting Fu <ting.fu@intel.com>
+Signed-off-by: Wenbin Chen <wenbin.chen@intel.com>
+---
+ libavfilter/dnn/dnn_backend_common.c   | 20 ++++++-
+ libavfilter/dnn/dnn_backend_common.h   |  3 +
+ libavfilter/dnn/dnn_backend_openvino.c |  2 +-
+ libavfilter/dnn/dnn_backend_tf.c       |  2 +-
+ libavfilter/dnn/dnn_backend_torch.cpp  | 33 +++++++++--
+ libavfilter/dnn_filter_common.c        | 78 +++++++++++++++++++++++---
+ libavfilter/dnn_filter_common.h        |  4 ++
+ libavfilter/dnn_interface.h            |  6 +-
+ libavfilter/vf_dnn_processing.c        |  3 +-
+ 9 files changed, 129 insertions(+), 22 deletions(-)
+
+diff --git a/libavfilter/dnn/dnn_backend_common.c b/libavfilter/dnn/dnn_backend_common.c
+index 91a4a3c4bf..766cb1480b 100644
+--- a/libavfilter/dnn/dnn_backend_common.c
++++ b/libavfilter/dnn/dnn_backend_common.c
+@@ -66,7 +66,10 @@ int ff_dnn_fill_task(TaskItem *task, DNNExecBaseParams *exec_params, void *backe
+     task->input_name = exec_params->input_name;
+     task->in_frame = exec_params->in_frame;
+     task->out_frame = exec_params->out_frame;
++    task->in_queue = exec_params->in_queue;
++    task->out_queue = exec_params->out_queue;
+     task->model = backend_model;
++    task->nb_input = exec_params->nb_input;
+     task->nb_output = exec_params->nb_output;
+     task->output_names = exec_params->output_names;
+ 
+@@ -153,8 +156,21 @@ DNNAsyncStatusType ff_dnn_get_result_common(Queue *task_queue, AVFrame **in, AVF
+ 
+     *in = task->in_frame;
+     *out = task->out_frame;
+-    ff_queue_pop_front(task_queue);
+-    av_freep(&task);
++    if (task->in_queue && task->out_queue) {
++        if (av_fifo_can_read(task->out_queue))
++            av_fifo_read(task->out_queue, out, 1);
++        if (av_fifo_can_read(task->in_queue))
++            av_fifo_read(task->in_queue, in, 1);
++        if (!av_fifo_can_read(task->out_queue) && !av_fifo_can_read(task->in_queue)) {
++            ff_queue_pop_front(task_queue);
++            av_fifo_freep2(&task->in_queue);
++            av_fifo_freep2(&task->out_queue);
++            av_freep(&task);
++        }
++    } else {
++        ff_queue_pop_front(task_queue);
++        av_freep(&task);
```

Comment:
task->in_queue or task->out_queue may not be NULL, is it required to call av_fifo_freep2() to release resource ?

Reply:
I change the conditional statement to "`(task->in_queue || task->out_queue)`", thanks.